ddl/db_test.go

```diff
@@ -1665,6 +1665,16 @@ func (s *testDBSuite) TestCreateTableWithPartition(c *C) {
 		partition p2 values less than maxvalue
 	);`)
 	c.Assert(ddl.ErrNotAllowedTypeInPartition.Equal(err), IsTrue)
+
+	// TODO: This SQL should return ErrPartitionFunctionIsNotAllowed: ERROR 1564 (HY000): This partition function is not allowed
```

Comment:
So, what does this PR solve? What will happen if we insert data into t9?

Reply:
will panic when creating t9.

Reply:
I mean after this PR.

Reply:
it will create table t9 successfully. 
But the correct behavior is return error: ERROR 1564 (HY000): This partition function is not allowed. 
I add a TODO in the comment and create an issue about this. issue#7058

Reply:
it will create table t9 successfully. 
But the correct behavior is return error: ERROR 1564 (HY000): This partition function is not allowed. 
I add a TODO in the comment and create an issue about this. issue#7058

Reply:
Why didn't you fix this bug in this PR?

Reply:
@tiancaiamao advise me to fix this bug in another PR.